package/run.sh

```diff
@@ -26,26 +26,25 @@ fi
 DELAY=10
 while ! env "INSTALL_K3S_FORCE_RESTART=${FORCE_RESTART}" "INSTALL_K3S_SKIP_DOWNLOAD=true" "INSTALL_K3S_SKIP_SELINUX_RPM=true" "INSTALL_K3S_SELINUX_WARN=true" installer.sh $@
 do
-    if ! systemctl cat k3s.service > /dev/null && echo $?; then
-        # if k3s.service can not be found then we are not a server node and thus will never encounter the 'too many learners' error, so we should just exit
-        echo "not a K3s server, not attempting error remediation"
-        exit 1
-    fi
-    # We need to get the right systemd service based off of the role we have, $INSTALL_K3S_EXEC will be set if we are a worker only node, in which case the systemd service is k3s-agent.service
-    journalctl -u k3s.service > k3s-service.txt
-    # Only use the logs for the latest restart of k3s, otherwise errors encountered after an ETCD join error will not be reported properly.
-    LAST_START_LINE=$(cat k3s-service.txt | grep "Starting k3s v1." -n  | cut -d: -f1 | tail -1)
-    if [[ "$LAST_START_LINE" -gt 0 ]]; then
-        LAST_LOGS=$(cat k3s-service.txt | sed -n "$((LAST_START_LINE))"',$p')
-        else
-        LAST_LOGS=$(cat k3s-service.txt)
-    fi
-    if ! echo "${LAST_LOGS}" | grep "ETCD join failed: etcdserver: too many learner members in cluster" -q && echo $?; then
-        exit 1
-        else
-        # We couldn't register as a learner, keep trying until we can
-        sleep "${DELAY}"
-    fi
+
+   # Get the last time the service started using the ExecMainStartTimestamp property
+   START=$(systemctl show k3s.service --property=ExecMainStartTimestamp | cut -f2 -d=)
+
+   # if START is n/a, we know we do not have access to the k3s.service and thus are
+   # running on a worker node which can not encounter the ETCD error, so we should just exit
+   if [ "$START" == "n/a" ]; then
```

Comment:
Hmm, I suspect you'll need to handle both `n/a` and an empty string; the `n/a` behavior only seems to occur on older versions of systemd.

```console
systemd-node-1:/ # systemctl show bogus.service --property=ExecMainStartTimestamp
ExecMainStartTimestamp=n/a
systemd-node-1:/ # systemctl show --property=Version
Version=249.11+suse.129.g17d488c53a
```

```console
brandond@dev01:~/go/src/github.com/k3s-io/k3s$ systemctl show bogus.service --property=ExecMainStartTimestamp
ExecMainStartTimestamp=
brandond@dev01:~/go/src/github.com/k3s-io/k3s$ systemctl show --property=Version
Version=251.4-1ubuntu7.1
```

Reply:
Okay, I added that additional condition. Also, I noticed that `systemctl show` has a `--value` flag that would allow us to drop the `cut -f2 -d=` part, but I wasn't confident that would be available in all versions - think its safe to use?

Reply:
Looks like it was added in https://github.com/systemd/systemd/commit/4f9a91055ce83be9b6a81bdeab6d360f13ff897a so systemd v230 or newer should have it. Unforunately EL7 is still on v219 and won't have it :/